### PR TITLE
make this repo support php 7.1

### PR DESCRIPTION
**Pull request by zhangjustin**
## Summary of changes
Update msgpack and zmq to support PHP 7.1.
@tjoelsson please review.

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -4,12 +4,40 @@
...
   changed_when: "'installed' not in pecl_msgpack_result.stdout"
   failed_when: "'ERROR' in pecl_msgpack_result.stdout"
+  when: pecl_zmq_version == '0.5.7'
 
 - name: ensure php zmq extension is installed
   shell: yes '' | pecl install zmq-{{ pecl_zmq_version }}
   register: pecl_zmq_result
   changed_when: "'installed' not in pecl_zmq_result.stdout"
   failed_when: "'ERROR' in pecl_zmq_result.stdout"
+  when: pecl_zmq_version == '1.1.2'
```
Comment (tjoelsson) [inline tasks/main.yml]: Only this version can be installed using pecl? Or it's because of PHP7?

Comment (zhangjustin) [reply tasks/main.yml]: When use PHP7.1 , can't use pecl to install zmq-1.1.2, zmq-1.1.3, 
when use php5,  we can use pecl to install zmq-1.1.2, zmq-1.1.3.
should be php7 cause this problem.

Comment (tjoelsson) [reply tasks/main.yml]: Okay, then let's use PHP version for this condition

File: defaults/main.yml [hunk 2]
```diff
@@ -3,3 +3,4 @@ pecl_msgpack_version: 0.5.7
 pecl_zmq_version: 1.1.2
 
 php_fpm_daemon: php-fpm
+php_version: 'php5'
```
Comment (tjoelsson) [inline defaults/main.yml]: better to check installed php version

Comment (tjoelsson) [reply defaults/main.yml]: or if it's too complicated, we can have variables zeromq_install_from_source and msgpack_install_from_source

File: tasks/main.yml [hunk 3]
```diff
@@ -1,43 +1,49 @@
...
   changed_when: "'installed' not in pecl_zmq_result.stdout"
   failed_when: "'ERROR' in pecl_zmq_result.stdout"
-  when: php_version == 'php5'
+  when: "'PHP 5' in php_version.stdout"
 
 - name: git clone zmq repo
   git:
     repo: "{{ zmq_source_repo }}"
     dest: "/tmp/php-zmq"
     version: "master"
```
Comment (tjoelsson) [inline tasks/main.yml]: make the version configurable (also for msgpack)

File: tasks/main.yml [hunk 4]
```diff
@@ -1,43 +1,49 @@
 ---
+- name: Register php version
+  shell: command -v php && php --version
+  register: php_version
+  changed_when: false
+  ignore_errors: true
```
Comment (tjoelsson) [inline tasks/main.yml:6]: maybe it should fail if php is not installed?

**Review by tjoelsson — CHANGES_REQUESTED** [review]

File: tasks/main.yml (hunk 4 above)
Comment (zhangjustin) [reply tasks/main.yml:6]: No, `fail` not working.

Comment (tjoelsson) [reply tasks/main.yml:6]: You mean it fails even if PHP is installed?

Comment (zhangjustin) [reply tasks/main.yml:6]: image
no matter php installed or not.

**Comment by zhangjustin** [comment]
Ready for review.

File: defaults/main.yml [hunk 5]
```diff
@@ -1,5 +1,9 @@
 ---
+msgpack_source_version: 'msgpack-2.0.2'
 pecl_msgpack_version: 0.5.7
 pecl_zmq_version: 1.1.2
 
 php_fpm_daemon: php-fpm
+zmq_source_repo: 'https://github.com/juwai/php-zmq.git'
+
+zmq_source_version: 'f2617063a4c007ca6073c0d09e9f36fd9b87ddaf'
```
Comment (tjoelsson) [inline defaults/main.yml]: Add a comment saying this is 1.1.3 (is it?)

File: defaults/main.yml [hunk 6]
```diff
@@ -1,5 +1,9 @@
 ---
+msgpack_source_version: 'msgpack-2.0.2'
 pecl_msgpack_version: 0.5.7
 pecl_zmq_version: 1.1.2
 
 php_fpm_daemon: php-fpm
+zmq_source_repo: 'https://github.com/juwai/php-zmq.git'
```
Comment (tjoelsson) [inline defaults/main.yml]: Remove

File: tasks/main.yml [hunk 7]
```diff
@@ -1,15 +1,49 @@
...
+  when: "'PHP 7' in php_version.stdout"
+
+- name: git clone msgpack repo
+  git:
+    repo: https://github.com/msgpack/msgpack-php.git
+    dest: "/tmp/php-msgpack"
+    version: "{{ msgpack_source_version }}"
+  when: "'PHP 7' in php_version.stdout"
+
+- name: install php-msgpack 1.1.3
```
Comment (tjoelsson) [inline tasks/main.yml]: "install msgpack php extension from source"

File: tasks/main.yml [hunk 8]
```diff
@@ -1,15 +1,49 @@
...
+    version: "{{ zmq_source_version }}"
+  when: "'PHP 7' in php_version.stdout"
+
+- name: install php-zmq 1.1.3
+  shell: phpize && ./configure && make && make install
+  args:
+    chdir: /tmp/php-zmq
+  when: "'PHP 7' in php_version.stdout"
+
+- name: git clone msgpack repo
```
Comment (tjoelsson) [inline tasks/main.yml]: msgpack-php

File: tasks/main.yml [hunk 9]
```diff
@@ -1,15 +1,49 @@
...
+  when: "'PHP 5' in php_version.stdout"
+
+- name: git clone zmq repo
+  git:
+    repo: https://github.com/mkoppanen/php-zmq.git
+    dest: "/tmp/php-zmq"
+    version: "{{ zmq_source_version }}"
+  when: "'PHP 7' in php_version.stdout"
+
+- name: install php-zmq 1.1.3
```
Comment (tjoelsson) [inline tasks/main.yml]: "install zmq php extension from source"

File: tasks/main.yml [hunk 10]
```diff
@@ -1,15 +1,49 @@
...
+  when: "'PHP 5' in php_version.stdout"
 
 - name: ensure php zmq extension is installed
   shell: yes '' | pecl install zmq-{{ pecl_zmq_version }}
   register: pecl_zmq_result
   changed_when: "'installed' not in pecl_zmq_result.stdout"
   failed_when: "'ERROR' in pecl_zmq_result.stdout"
+  when: "'PHP 5' in php_version.stdout"
+
+- name: git clone zmq repo
```
Comment (tjoelsson) [inline tasks/main.yml]: php-zmq

**Review by tjoelsson — CHANGES_REQUESTED** [review]

**Comment by zhangjustin** [comment]
Updated, please check.

**Review by tjoelsson — APPROVED** [review]

**Comment by zhangjustin** [comment]
@imlazyone , please review this PR.

**Comment by tjoelsson** [comment]
@mark-juwai can review

**Review by mark-juwai — APPROVED** [review]

**Pull request merged**